### ISSUE TITLE
how many memory do one message take up 

### ISSUE THREAD

**Issue by lougxing** [nsqio/nsq#1168]
One topic have three channcels. A producter product one message,  three channel haven't consumers.  if the message is 1024K,   Will every channel keep one copy ?  
eg, Is the memory 3*1024K ?

**Comment by mdh67899** [comment]
Yes, if you produce one message to topic, topic will send a copy to each of the 3 channels
 if channel's memoryMsgChan is full, channel will write message to his local disk files

https://github.com/nsqio/nsq/blob/1db3903c4f9c0ca40a3ef35b89102f43dca30761/nsqd/topic.go#L310-L325

**Comment by ploxiln** [comment]
there will be three message struct copies, which are somewhere around 96 bytes each:
https://github.com/nsqio/nsq/blob/1db3903c4f9c0ca40a3ef35b89102f43dca30761/nsqd/message.go#L18-L30

Each has a slice for the body which I think works as a reference, so I think there will be just one copy of the 1024K body between all three messages.
